oak_restricted_kernel/src/lib.rs

```diff
@@ -282,6 +282,64 @@ pub fn start_kernel(info: &BootParams) -> ! {
     let heap_page_range = VMA_ALLOCATOR.lock().allocate(1 << 19).unwrap();
     memory::init_kernel_heap(heap_page_range).unwrap();
 
+    let _stage0_dice_data = {
+        let dice_memory_slice = {
+            let e820_dice_data_entry = info
+                .e820_table()
+                .iter()
+                .find(|e| e.entry_type() == Some(oak_linux_boot_params::E820EntryType::DiceData))
+                .expect("failed to find dice data");
+
+            let phys_start_addr = PhysAddr::new_truncate(
+                e820_dice_data_entry
+                    .addr()
+                    .try_into()
+                    .expect("couldn't convert usize to u64"),
+            );
+
+            // Validate that the dice data mem address matches the kernel args if present
+            if let Some(arg) = kernel_args.get(&alloc::format!(
+                "--{}",
+                oak_dice::evidence::DICE_DATA_CMDLINE_PARAM
+            )) {
+                let parsed_arg = u64::from_str_radix(
+                    arg.strip_prefix("0x")
+                        .expect("failed stripping the hex prefix"),
+                    16,
+                )
+                .expect("couldn't parse address as a hex number");
+                if parsed_arg != phys_start_addr.as_u64() {
+                    panic!("inconsistent dice data addresses supplied in the E820 table and kernel args")
+                }
+            }
+
+            let virt_start_addr = {
+                let pt = PAGE_TABLES.get().expect("failed to get page tables");
+                pt.translate_physical(phys_start_addr)
+                    .expect("failed to translate physical dice address")
+            };
+
+            // Safety: the E820 table indicated that this is the corrct memory segment.
+            unsafe {
+                core::slice::from_raw_parts_mut::<u8>(
+                    virt_start_addr.as_mut_ptr(),
+                    e820_dice_data_entry.size(),
+                )
+            }
+        };
```

Comment:
Eyeballing this looks similar to the logic above -- I haven't looked at the details, but is it not possible to share at least some parts? Or if it's not possible, at least add a comment to keep them in sync manually.

Reply:
good catch, it should not be duplicated. I made mistake when moving it, and forgot to delete it in its previous location.